src/target/source/codegen_opencl.cc

```diff
@@ -541,12 +541,26 @@ void CodeGenOpenCL::VisitExpr_(const OrNode* op, std::ostream& os) {
 }
 
 void CodeGenOpenCL::VisitExpr_(const SelectNode* op, std::ostream& os) {
+  std::ostringstream oss;
   os << "select(";
-  PrintExpr(op->false_value, os);
+  PrintExpr(op->false_value, oss);
+  os << CastFromTo(oss.str(), op->false_value.dtype(), op->dtype);
+  oss.str("");
   os << ", ";
-  PrintExpr(op->true_value, os);
+  PrintExpr(op->true_value, oss);
+  os << CastFromTo(oss.str(), op->true_value.dtype(), op->dtype);
+  oss.str("");
   os << ", ";
-  PrintExpr(op->condition, os);
+  PrintExpr(op->condition, oss);
+  if (op->dtype.is_float()) {
```

Comment:
I'm not sure, but maybe checking the type of `op->condition` will be enough?
I mean only: 
```
    if (op->condition.dtype().is_uint() || op->condition.dtype().is_int()) {
      os << oss.str();
    } else {
      os << CastTo(oss.str(), DataType::Int(op->dtype.bits(), op->dtype.lanes()));
    }
```

Reply:
According to the header, we should only cast the condition operand to int when the return type is a floating point type.

Cond should be casted to the same type with both the true and false operands when not selecting from float.

Reply:
According to the header, we should only cast the condition operand to int when the return type is a floating point type.

Cond should be casted to the same type with both the true and false operands when not selecting from float.

Reply:
Sorry, my bad. I wasn't careful enough when I read the header file. Thank you for your clarification. I double-checked the header and I agree with you. Now, this if condition is clear enough for me.